File: 300 Longest Increasing Subsequence/Longest Increasing Subsequence.js
var lengthOfLIS = function (nums) {
  const dp = new Array(nums.length).fill(1);

  for (let i = 1; i < nums.length; i++) {
    for (let j = 0; j < i; j++) {
      if (nums[j] < nums[i]) {
        dp[i] = Math.max(dp[i], dp[j] + 1);
      }
    }
  }

  let maxLength = 0;
  for (let length of dp) {
    maxLength = Math.max(maxLength, length);
  }

  return maxLength;
};

File: 300 Longest Increasing Subsequence/Longest Increasing Subsequence.cpp
class Solution {
public:
    int lengthOfLIS(vector<int>& nums) {
        vector<int> dp(nums.size(), 1);
        
        for (int i = 1; i < nums.size(); i++) {
            for (int j = 0; j < i; j++) {
                if (nums[j] < nums[i]) {
                    dp[i] = max(dp[i], dp[j] + 1);
                }
            }
        }
        
        int maxLength = 0;
        for (int length : dp) {
            maxLength = max(maxLength, length);
        }
        
        return maxLength;
    }
};
